src/assets/wise5/authoringTool/edit-branch/edit-branch.component.ts

```diff
@@ -0,0 +1,260 @@
+        createBranchCriterion = 'workgroupId';
+      } else if (this.node.transitionLogic.howToChooseAmongAvailablePaths === 'random') {
+        createBranchCriterion = 'random';
+      } else if (this.node.transitionLogic.howToChooseAmongAvailablePaths === 'tag') {
+        createBranchCriterion = 'tag';
+      }
+    }
+    this.setCriteria(createBranchCriterion);
+    if (createBranchNodeId != null) {
+      this.setNodeId(createBranchNodeId);
+    }
+    if (createBranchComponentId) {
+      this.setComponentId(createBranchComponentId);
+    }
+
+    // use set timeout to give the form time to populate the path form group controls
+    setTimeout(() => {
+      if (scores.length > 0) {
+        for (let i = 0; i < scores.length; i++) {
+          this.pathFormGroup.controls[`path-${i + 1}`].setValue(scores[i].join(', '));
+        }
+      }
+      if (choices.length > 0) {
+        for (let i = 0; i < choices.length; i++) {
+          this.pathFormGroup.controls[`path-${i + 1}`].setValue(choices[i]);
+        }
+      }
+    });
+    this.changeDetector.detectChanges();
+  }
+
+  protected removeBranchButtonClicked(): void {
+    if (
+      confirm(
+        $localize`Are you sure you want to remove the branch?\n\nThe branch structure will be removed but the steps will not be deleted.`
+      )
+    ) {
+      this.removeBranch();
+    }
+  }
+
+  private removeBranch(): void {
+    for (let bp = 0; bp < this.branchPaths.length; bp++) {
+      const branchPath = this.branchPaths[bp];
+      this.removeBranchPath(branchPath);
+      bp--; // shift the counter back one because we have just removed a branch path
+    }
+
+    const nodeId = this.targetId; // branch point node
+    const nodeIdAfter = this.projectService.getNodeIdAfter(nodeId);
+
+    /*
+     * update the transition of this step to point to the next step
+     * in the project. this may be different than the next step
+     * if it was still the branch point.
+     */
+    this.projectService.setTransition(nodeId, nodeIdAfter);
+
+    this.projectService.setTransitionLogicField(nodeId, 'howToChooseAmongAvailablePaths', null);
+    this.projectService.setTransitionLogicField(nodeId, 'whenToChoosePath', null);
+    this.projectService.setTransitionLogicField(nodeId, 'canChangePath', null);
+    this.projectService.setTransitionLogicField(nodeId, 'maxPathsVisitable', null);
+
+    /*
+     * branch paths are determined by the transitions. since there is now
+     * just one transition, we will create a single branch object to
+     * represent it.
+     */
+
+    // create a branch object to hold all the related information for that branch
+    const branch: any = {
+      number: 1
+    };
+
+    /*
+     * set the mapping of all the ids to order for use when choosing which items are
+     * in the branch path
+     */
+    branch.items = this.getBranchItems();
+    branch.checkedItemsInBranchPath = [];
```

Comment:
Checked items doesn't make sense here because there's nothing the author checks off. Should this instead be nodesInBranchPath?
```suggestion
    branch.nodesInBranchPath = [];
```